src/plugins/dashboard/public/services/dashboard_content_management/lib/load_dashboard_state.ts

```diff
@@ -58,14 +59,28 @@ export const loadDashboardState = async ({
   /**
    * Load the saved object from Content Management
    */
-  const { item: rawDashboardContent, meta: resolveMeta } = await contentManagement.client
-    .get<DashboardCrudTypes['GetIn'], DashboardCrudTypes['GetOut']>({
-      contentTypeId: DASHBOARD_CONTENT_ID,
-      id,
-    })
-    .catch((e) => {
-      throw new SavedObjectNotFound(DASHBOARD_CONTENT_ID, id);
-    });
+  let rawDashboardContent;
+  let resolveMeta;
+
+  const cachedDashboard = dashboardContentManagementCache.fetchDashboard(id);
```

Comment:
We unfortunately can't use the dashboard service `findDashboardById` here because it does a bunch of type wrapping and removal of "unnecessary" data for consumers 😅 And we need the native type with `meta` and such in the dashboard state.